var/spack/repos/builtin/packages/dbow2/package.py

```diff
@@ -19,3 +19,4 @@ class Dbow2(CMakePackage):
     depends_on('opencv+calib3d+core+features2d+highgui+imgproc')
     depends_on('boost')
     depends_on('dlib')
+    depends_on('eigen', type='build')
```

Comment:
In our current model `eigen` would be a `link` type dependency (`build` dependencies are basically build tools required in the PATH at build time while `eigen` is a header only library)
```suggestion
    depends_on('eigen', type='link')
```